File: frontend/Haocean-Mobile/api/record.js
import request from '@/utils/request'

// 查询执法记录列表
export function listRecord(query) {
  return request({
    url: '/system/record/list',
    method: 'get',
    params: query
  })
}

// 获取执法记录详情
export function getRecord(recordId) {
  return request({
    url: '/system/record/' + recordId,
    method: 'get'
  })
}

// 新增执法记录
export function addRecord(data) {
  return request({
    url: '/system/record',
    method: 'post',
    data: data
  })
}

// 修改执法记录
export function updateRecord(data) {
  return request({
    url: '/system/record',
    method: 'put',
    data: data
  })
}

// 删除执法记录
export function deleteRecord(recordIds) {
  return request({
    url: '/system/record/' + recordIds,
    method: 'delete'
  })
}

// 上报执法记录
export function submitRecord(recordId) {
  return request({
    url: '/system/record/submit/' + recordId,
    method: 'put'
  })
}

// 上传证据
export function uploadEvidence(data) {
  return request({
    url: '/system/evidence/upload',
    method: 'post',
    formData: true
  })
}

// 删除证据
export function deleteEvidence(evidenceId) {
  return request({
    url: '/system/evidence/' + evidenceId,
    method: 'delete'
  })
}

File: app/admin/router/system/record.go
package system

import (
	"github.com/gin-gonic/gin"
	"haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitRecord 初始化执法记录路由
func InitRecord(e *gin.Engine) {
	v := e.Group("system")
	{
		v.GET("/record/list", system.ListRecord)
		v.GET("/record/:id", system.GetRecord)

		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			auth.POST("/record", system.AddRecord)
			auth.PUT("/record", system.UpdateRecord)
			auth.DELETE("/record/:ids", system.DeleteRecord)
			auth.PUT("/record/submit/:id", system.SubmitRecord)
			auth.POST("/evidence/upload", system.UploadEvidence)
			auth.DELETE("/evidence/:id", system.DeleteEvidence)
		}
	}
}
